src/System.Linq/src/System/Linq/AppendPrepend.cs

```diff
@@ -272,6 +293,11 @@ private SingleLinkedNode(SingleLinkedNode<TSource> linked, TSource item)
 
             public IEnumerator<TSource> GetEnumerator()
             {
+                return ToArray().AsEnumerable().GetEnumerator();
+            }
+
+            public TSource[] ToArray()
```

Comment:
Why separate this out into its own (public) method?  Where else is it being used?